packages/database/src/api/Query.ts

```diff
@@ -295,6 +299,17 @@ export class Query {
     this.repo.removeEventCallbackForQuery(this, container);
   }
 
+  get(): Promise<DataSnapshot> {
+    const deferred = new Deferred<DataSnapshot>();
+    const callback = (snapshot: DataSnapshot) => {
+      deferred.resolve(snapshot);
+    };
+    this.repo.get(this, callback, err => {
+      deferred.reject(err);
+    });
```

Comment:
Do you think you could push the callback to Promise conversion further down? The existing code uses a lot of callbacks since Promises can't drive Event-based logic, but the code that you are adding in this PR should be able to rely mostly on Promises.